lib/Query/Resolver.php

```diff
@@ -298,7 +298,9 @@ protected function addForeignKeys(Table $table)
                     continue;
                 }
 
-                $foreignTable = $relation->mapper()->getMapper($relation->entityName())->table();
+                $foreignTableMapper = $relation->mapper()->getMapper($relation->entityName());
+                $foreignTableMapper->migrate();
```

Comment:
I was testing your fixes on my fork and ran into a recursion issue. My entity is referencing itself (hierarchical relation). 
This line creates an infinite loop resulting in a Maximum nesting level reached error. Commenting this line fixes the problem. Is there a reason why it was added?